The-Orm/SynthBankPanel.cpp

```diff
@@ -45,23 +45,29 @@ SynthBankPanel::SynthBankPanel(midikraft::PatchDatabase& patchDatabase, PatchVie
 	sendButton_.setButtonText("Send to synth");
 	sendButton_.onClick = [this]() {
 		if (patchView_ && synthBank_) {
+			auto activeSynthBank = synthBank_;
+			
 			if (isUserBank()) {
-				patchView_->sendBankToSynth(synthBank_, true,  []() {
-					spdlog::info("Bank sent successfully!");
-				});
-			} 
-			else
-			{
-				patchView_->sendBankToSynth(synthBank_, false, [this]() {
-					// Save it in the database now that we have successfully sent it to the synth
-					patchDatabase_.putPatchList(synthBank_);
-					// Mark the bank as not modified
-					synthBank_->clearDirty();
-					refresh();
-				});
+				// Retrieve the corresponding active synth bank
+				auto activeSynthBankId = midikraft::ActiveSynthBank::makeId(synthBank_->synth(), synthBank_->bankNumber());
+				midikraft::ListInfo info = {activeSynthBankId, ""};
+				auto activeSynthBank = std::dynamic_pointer_cast<midikraft::SynthBank>(patchView_->retrieveListFromDatabase(info));
+				if(!activeSynthBank)
+					return;
```

Comment:
Currently relies on the user having loaded the corresponding synth bank first.

Instead of aborting with an error, we could call "Librarian::startDownloadingAllPatches" for that bank and perform the update in the finished callback. Thoughts?

Reply:
if I understand correctly you force a download of the bank first in case it hasn't been downloaded? That would work, my main concern was that you could accidentally overwrite patches in the synth that have never been downloaded, so you'd have no backup of them.

As some synths are really slow to download, the user could be surprised in that a lengthy download operation is started, but I think as this is abortable that's probably ok.